File: examples/conditions-operator-if-exists/conditions-operator-if-exists.py
from iam_floyd import statement, Operator
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.Ec2() \
        .allow() \
        .to_start_instances() \
        .if_aws_request_tag('Environment',
                            ['Production', 'Staging', 'Dev'],
                            Operator().string_equals().if_exists())
    # doc-end
)

all = [s]
out(all)
deploy(all)

File: examples/conditions-operator-if-exists/conditions-operator-if-exists.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Operator, Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.Ec2()
      .allow()
      .toStartInstances()
      .ifAwsRequestTag(
        'Environment',
        ['Production', 'Staging', 'Dev'],
        new Operator().stringEquals().ifExists(),
      )
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);
